packages/graphiql-create-fetcher/src/lib.ts

```diff
@@ -124,24 +127,40 @@ export const createLegacyWebsocketsFetcher = (
  */
 export const createMultipartFetcher = (
   options: CreateFetcherOptions,
-): Fetcher => async (graphQLParams: FetcherParams, fetcherOpts?: FetcherOpts) =>
-  makeAsyncIterableIteratorFromSink<FetcherResult>(sink => {
-    fetchMultipart<FetcherResult>(options.url, {
+  httpFetch: typeof fetch,
+): Fetcher =>
+  async function* (graphQLParams: FetcherParams, fetcherOpts?: FetcherOpts) {
+    const response = await httpFetch(options.url, {
       method: 'POST',
       body: JSON.stringify(graphQLParams),
       headers: {
         'content-type': 'application/json',
+        accept: 'application/json, multipart/mixed',
         ...options.headers,
         // allow user-defined headers to override
         // the static provided headers
         ...fetcherOpts?.headers,
       },
-      onNext: parts => {
-        // @ts-ignore
-        sink.next(parts);
-      },
-      onError: sink.error,
-      onComplete: sink.complete,
+    }).then(response => {
+      // TODO: Can we make this payload type-safe?
```

Comment:
I think we would expect `FetcherResult` here right?

Reply:
ya, right!